xds/src/main/java/com/linecorp/armeria/xds/SotwXdsStream.java

```diff
@@ -118,28 +118,6 @@ public void stop(Throwable throwable) {
         requestObserver = null;
     }
 
-    private void retryOrClose(Status status) {
```

Comment:
> In responseHandler.handleReset(this);, all resources are updated even if this stream is for a resource. Is this intentional?

Good point, I missed this 😅   I added a `targetTypes` parameter to `SotwXdsStream` which filters which resource types a stream targets

> How about also calling subscriberStorage...(this::resourcesUpdated); directly instead of calling responseHandler.handleReset(this); because calling resourceUpdated is somewhat not related to the XdsResponseHandler?

Sure, updated